src/hooks/useFormField.ts

```diff
@@ -36,6 +36,8 @@ export const useFormField = <
     minDate,
   }: UseFieldConfig<FieldValue, InputValue> & ValidatorProps,
 ) => {
+  const serializedRegex = useMemo(() => regex?.toString(), [regex])
```

Comment:
Why do we use `toString` here?

Reply:
The whole idea is to serialize `regex` prop to a primitive so that it is memoizable.

In this case, we want to transform `RegExp[] | RegExp[][]` to `string`.

Turns out `toString()` method is super powerful, it will flatten the regexes and join them with a comma 👌

Reply:
Makes sense, thanks!